rewrite-gradle/src/main/java/org/openrewrite/gradle/UpdateJavaCompatibility.java

```diff
@@ -135,6 +149,56 @@ private G.CompilationUnit addCompatibilityTypeToSourceFile(G.CompilationUnit c,
             }
             return c;
         }
+
+        private G.CompilationUnit addJavaToolchainToSourceFile(G.CompilationUnit c, ExecutionContext ctx) {
+            Optional<J.MethodInvocation> javaStatement = c.getStatements().stream().filter(s -> s instanceof J.MethodInvocation)
+                    .map(s -> (J.MethodInvocation) s)
+                    .filter(m -> m.getSimpleName().equals("java"))
+                    .findFirst();
+
+            if (javaStatement.isPresent()) {
+                List<Expression> args = javaStatement.get().getArguments();
+                J.Lambda jLambda = (J.Lambda) args.get(0);
+                J.Block body = (J.Block) jLambda.getBody();
+                Statement toolchainStatement = GradleParser.builder().build()
+                        .parse(ctx, "\ntoolchain {" +
+                            "\n  languageVersion = JavaLanguageVersion.of(" + version + ")" +
+                            "\n}")
+                        .map(G.CompilationUnit.class::cast)
+                        .findFirst()
+                        .orElseThrow(() -> new IllegalArgumentException("Could not parse as Gradle"))
+                        .getStatements()
+                        .get(0);
+                J.Lambda newLambda = jLambda.withBody(body.withStatements(ListUtils.concat(toolchainStatement, body.getStatements())));
+                List<Expression> newArgs = new ArrayList<>();
+                newArgs.add(newLambda);
+                J.MethodInvocation newJavaStatement = javaStatement.get().withArguments(newArgs);
+                List<Statement> newStatements = new ArrayList<>();
+                for (Statement s : c.getStatements()) {
+                    if (s instanceof J.MethodInvocation) {
+                        J.MethodInvocation invocation = (J.MethodInvocation) s;
+                        if (invocation.getName().getSimpleName().equals("java")) {
+                            newStatements.add(newJavaStatement);
+                            continue;
+                        }
+                    }
+                    newStatements.add(s);
+                }
+                c = c.withStatements(newStatements);
```

Comment:
Maybe we use `ListUtils.concat(before, toolchain)` like we do on the other side. It's a little more idiomatic that way.